File: server/lib/util/school.ts
import { School } from '@/generated/api/v1/user'
import { School as PrismaSchool } from '@prisma/client'

export function protoSchoolToPrismaSchool(school: School): PrismaSchool {
  const map = {
    [School.SCHOOL_AAHS]: PrismaSchool.AAHS,
    [School.SCHOOL_AIT]: PrismaSchool.AIT,
    [School.SCHOOL_APA]: PrismaSchool.APA,
    [School.SCHOOL_MHS]: PrismaSchool.MHS,
    [School.SCHOOL_UCTECH]: PrismaSchool.UCTECH,
  } satisfies Partial<Record<School, PrismaSchool>>

  if (school in map) {
    return map[school as keyof typeof map]
  }

  throw new Error(`Invalid school: ${school}`)
}

export function prismaSchoolToProtoSchool(school: PrismaSchool): School {
  const map = {
    [PrismaSchool.AAHS]: School.SCHOOL_AAHS,
    [PrismaSchool.AIT]: School.SCHOOL_AIT,
    [PrismaSchool.APA]: School.SCHOOL_APA,
    [PrismaSchool.MHS]: School.SCHOOL_MHS,
    [PrismaSchool.UCTECH]: School.SCHOOL_UCTECH,
  } satisfies Partial<Record<PrismaSchool, School>>

  if (school in map) {
    return map[school as keyof typeof map]
  }

  throw new Error(`Invalid school: ${school}`)
}

File: app/src/main/java/lol/calico/uctutors/util/School.kt
package lol.calico.uctutors.util

import lol.calico.uctutors.generated.api.v1.School

fun schoolFrom(school: String): School =
  when (school) {
    "AAHS" -> School.SCHOOL_AAHS
    "AIT" -> School.SCHOOL_AIT
    "APA" -> School.SCHOOL_APA
    "MHS" -> School.SCHOOL_MHS
    "UCTECH",
    "UCT" -> School.SCHOOL_UCTECH
    else -> throw IllegalArgumentException("Invalid school $school")
  }

fun schoolToString(school: School): String =
  when (school) {
    School.SCHOOL_AAHS -> "AAHS"
    School.SCHOOL_AIT -> "AIT"
    School.SCHOOL_APA -> "APA"
    School.SCHOOL_MHS -> "MHS"
    School.SCHOOL_UCTECH -> "UCTECH"
    else -> ""
  }
